pio/lib/Sender/Sender.cpp

```diff
@@ -485,3 +487,237 @@ bool SenderClass::sendTCONTROL(String server, uint16_t port)
+
+        for (const auto &kv : _doc.as<JsonObject>())
+        {
+            _pin = atoi(kv.key().c_str());
+            _value = kv.value().as<String>();
+            Blynk.virtualWrite(_pin, _value);
+        }
+    }
+
+    else {
+        CONSOLELN(F("\nFailed to connect to Blynk, going to sleep"));
+        return false;
+    }
+    
+    delay(300);     //delay to allow last value to be sent;
+    return true;
+}
+
+bool SenderClass::sendBlynkHTTP(String token)
+{
+    //http://blynk-cloud.com/auth_token/update/pin
+    serializeJson(_doc, Serial);
+
+    String msg = String("PUT /");
+    msg += token;
+    msg += "/update/V";
+
+    String _msg;
+    String _value;
+    String _key;
+
+    //Blynk does not allow multiple pins write via POST, PUT or GET, therefore we ned to send multiple writes.
+    for (const auto &kv : _doc.as<JsonObject>())
+    {
+        if (_client.connect(BLYNKSERVER, 80))
+        {
+            CONSOLELN(F("\nConnection OK, Sender: Blynk PUT"));
+
+            _msg = msg;
+
+            _msg += String(kv.key().c_str());
+            _value = String("[\"") + kv.value().as<String>() + "\"]";
+            _client.print(_msg);
+
+            //https://github.com/blynkkk/blynk-library/blob/2013a8b86e5a429532ab9fe3a7fe7c2dbe9b0536/examples/Boards_With_HTTP_API/ESP8266/ESP8266.ino#L72
+            _client.println(F(" HTTP/1.1"));
+            _client.print(F("Host: ")); 
+            _client.println(BLYNKSERVER);
+            _client.println(F("Connection: close"));
+            if (_value.length()) {
+                _client.println(F("Content-Type: application/json"));
+                _client.print(F("Content-Length: ")); _client.println(_value.length());
+                _client.println();
+                _client.print(_value);
+            } else {
+                _client.println();
+            }
+
+            int timeout = millis() + CONNTIMEOUT;
+            while (_client.available() == 0) {
+                if (timeout - millis() < 0) {
+                    CONSOLELN(F("\n>>> Client Timeout !"));
+                    _client.stop();
+                    return false;
+                }
+            }
+
+            while (_client.available())
+            {
+                char c = _client.read();
+                Serial.write(c);  
+            }
+    
+            _client.stop();
+        }
+        else{
+            CONSOLELN(F("\nERROR Sender: couldnt connect"));
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void SenderClass::sendBlynkEmail(String Subject, String Content)
+{
+    uint32_t _validflag;
+    ESP.rtcUserMemoryRead(RTCEMAILADDR, &_validflag, sizeof(_validflag));
+
+    if (_validflag != RTCVALIDFLAG)
+    {
+      Blynk.email(Subject, Content);
+      _validflag = RTCVALIDFLAG;
+      ESP.rtcUserMemoryWrite(RTCEMAILADDR, &_validflag, sizeof(_validflag));
+      delay(200);
+    }
+}
+
+uint8_t SenderClass::blynkTerminal(String token)
```

Comment:
I am not in favor for another configuration procedure besides of Wifi Manager UI. It is narrow minded, possibly fitting to Blynk but not at all for general purpose. It’s a scope creep

Reply:
`::blynkTerminal` ? It's not fully implemented yet on this PR... The idea behind it is to trigger the configuration without opening the container.

We can simply remove it without affecting the current functionality, please let me know.

Reply:
Yes please remove it then

Reply:
ok